## 86% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 1
Expected to fail tests: 0
Passed tests: 6
```
```
IDs of failed tests: 703133 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=feat/swap-set-max-slippage&PR_ID=21295&APK_NAME=21295.apk&TR_CASE_IDS=703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687002">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:243: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5f5cdb34318c4367aeeda6950d4107f2?auth=7b90d44bd6f365fee217d3ba5c0a7be0#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687004">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687003">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687000">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3c176879a68f42a29d68aaa4291694c8?auth=f31701af826cc943d10649b4cf0c2cf6#99">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2cc25206add9486aa7f4fefc5069026d?auth=12c3449f5011222a7dd507c5830c9fc9#90">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687005">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9df4810ff84949a38b8ec1439ab1e22b?auth=3b7cc75b7d0bcddbb90cbe2d1c3425e2#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3686999">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5f5cdb34318c4367aeeda6950d4107f2?auth=7b90d44bd6f365fee217d3ba5c0a7be0#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3687001">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ae018743f39a437b82b702f0ec5ca6bf?auth=c9be27e9ebcc6d331230eb295868ec10#162">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6260a305c9064da2a2510102d96205d4?auth=8c23f0800c8197586576b9630181ae43#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/quo/components/settings/data_item/view.cljs
+++ b/src/quo/components/settings/data_item/view.cljs
@@ -148,7 +148,7 @@
                      colors/neutral-50)]
     [rn/pressable
      {:accessibility-label :data-item
-      :disabled            (not right-icon)
+      :disabled            (nil? on-press)
       :on-press            on-press
       :style               (merge (style/container {:size        size
                                                     :card?       card?

--- a/src/status_im/constants.cljs
+++ b/src/status_im/constants.cljs
@@ -591,6 +591,7 @@
 (def ^:const slippages [0.1 0.5 1])
 (def ^:const default-slippage 0.5)
 (def ^:const max-recommended-slippage 5)
+(def ^:const max-slippage 30)
 (def ^:const max-slippage-decimal-places 2)
 (def ^:const swap-provider-paraswap
   {:name                     :paraswap

--- a/src/status_im/contexts/wallet/sheets/slippage_settings/view.cljs
+++ b/src/status_im/contexts/wallet/sheets/slippage_settings/view.cljs
@@ -2,18 +2,23 @@
   (:require [clojure.string :as string]
             [quo.core :as quo]
             [react-native.core :as rn]
+            [react-native.platform :as platform]
             [status-im.constants :as constants]
             [status-im.contexts.wallet.sheets.slippage-settings.style :as style]
             [utils.i18n :as i18n]
-            [utils.number]
+            [utils.money :as money]
+            [utils.number :as number]
             [utils.re-frame :as rf]))
 
 (defn- validate-slippage
   [slippage]
-  (let [slippage-value (utils.number/parse-float slippage)]
+  (let [slippage-value (number/parse-float slippage)]
     (cond
       (<= slippage-value 0)
       {:message (i18n/label :t/slippage-should-be-more-than-0)
+       :type    :error}
+      (> slippage-value constants/max-slippage)
+      {:message (i18n/label :t/slippage-cant-be-more-than-30)
        :type    :error}
       (> (count (second (string/split slippage ".")))
          constants/max-slippage-decimal-places)
@@ -33,14 +38,37 @@
     (subs s 0 (dec (count s)))
     (str s k)))
 
+(defn- calculate-receive-amount
+  [asset-to-receive amount-out max-slippage]
+  (let [receive-token-decimals    (:decimals asset-to-receive)
+        amount-out-whole-number   (number/hex->whole amount-out receive-token-decimals)
+        amount-out-num            (number/to-fixed amount-out-whole-number
+                                                   (min constants/min-token-decimals-to-display
+                                                        receive-token-decimals))
+        slippage-num              (-> (money/bignumber amount-out-num)
+                                      (money/mul (money/bignumber max-slippage))
+                                      (money/div (money/bignumber 100)))
+        amount-out-minus-slippage (money/sub (money/bignumber amount-out-num) slippage-num)
+        receive-amount            (if (money/greater-than amount-out-minus-slippage
+                                                          (money/bignumber 0))
+                                    (str amount-out-minus-slippage)
+                                    0)]
+    receive-amount))
+
 (defn view
   []
   (let [current-slippage                (rf/sub [:wallet/swap-max-slippage])
         account-color                   (rf/sub [:wallet/current-viewing-account-color])
+        asset-to-receive                (rf/sub [:wallet/swap-asset-to-receive])
+        amount-out                      (rf/sub [:wallet/swap-proposal-amount-out])
         [max-slippage set-max-slippage] (rn/use-state (str current-slippage))
         [error set-error]               (rn/use-state nil)
         [custom? set-custom?]           (rn/use-state (not-any? #{current-slippage}
                                                                 constants/slippages))
+        receive-token-symbol            (:symbol asset-to-receive)
+        receive-amount                  (calculate-receive-amount asset-to-receive
+                                                                  amount-out
+                                                                  max-slippage)
         handle-slippage-change          (rn/use-callback
                                          (fn [value]
                                            (let [new-slippage (update-string-on-keypress value
@@ -50,7 +78,8 @@
                                          [max-slippage set-max-slippage set-error])
         on-select-slippage              (rn/use-callback (fn [slippage]
                                                            (set-max-slippage (str slippage))
-                                                           (set-custom? (not slippage)))
+                                                           (set-custom? (nil? slippage))
+                                                           (when slippage (set-error nil)))
                                                          [set-max-slippage set-custom?])
         save-disabled?                  (rn/use-memo (fn []
                                                        (or (= max-slippage (str current-slippage))
@@ -73,19 +102,23 @@
              [quo/drawer-action
               (cond-> {:title    (str slippage "%")
                        :on-press #(on-select-slippage slippage)}
-                (= (str slippage) max-slippage) (assoc :state :selected))])
+                (and (= (str slippage) max-slippage) (not custom?)) (assoc :state :selected))])
            constants/slippages)
       [quo/drawer-action
        (cond-> {:title       (i18n/label :t/custom)
                 :action      :input
                 :on-press    #(on-select-slippage nil)
-                :input-props {:auto-focus          true
-                              :customization-color account-color
-                              :placeholder         (i18n/label :t/type-slippage)
-                              :right-icon          {:icon-name :i/percentage
-                                                    :on-press  identity
-                                                    :style-fn  style/percentage-icon}
-                              :value               max-slippage}}
+                :input-props {:auto-focus               true
+                              :customization-color      account-color
+                              :placeholder              (i18n/label :t/type-slippage)
+                              :right-icon               {:icon-name :i/percentage
+                                                         :on-press  identity
+                                                         :style-fn  style/percentage-icon}
+                              :value                    max-slippage
+                              :show-soft-input-on-focus false
+                              :on-focus                 (fn []
+                                                          (when platform/android?
+                                                            (rf/dispatch [:dismiss-keyboard])))}}
          custom? (assoc :state :selected))]]
      (when (and custom? error)
        [quo/info-message
@@ -107,8 +140,8 @@
        :description          :top
        :context-tag-props    {:size   24
                               :type   :token
-                              :token  "USDT"
-                              :amount "99.97"} ;; will be replaced with real data later
+                              :token  receive-token-symbol
+                              :amount receive-amount}
        :description-top-text (i18n/label :t/receive-at-least)}]
      (when custom?
        [quo/numbered-keyboard

--- a/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
+++ b/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
@@ -12,6 +12,7 @@
             [status-im.contexts.wallet.common.account-switcher.view :as account-switcher]
             [status-im.contexts.wallet.common.utils :as utils]
             [status-im.contexts.wallet.sheets.buy-token.view :as buy-token]
+            [status-im.contexts.wallet.sheets.slippage-settings.view :as slippage-settings]
             [status-im.contexts.wallet.swap.setup-swap.style :as style]
             [status-im.contexts.wallet.swap.utils :as swap-utils]
             [utils.debounce :as debounce]
@@ -39,7 +40,7 @@
                   {:clean-approval-transaction? clean-approval-transaction?}])))
 
 (defn- data-item
-  [{:keys [title subtitle size subtitle-icon subtitle-color loading?]}]
+  [{:keys [title subtitle size subtitle-icon subtitle-color on-press loading?]}]
   [quo/data-item
    {:container-style style/detail-item
     :blur?           false
@@ -50,7 +51,8 @@
     :subtitle        subtitle
     :size            size
     :icon            subtitle-icon
-    :subtitle-color  subtitle-color}])
+    :subtitle-color  subtitle-color
+    :on-press        on-press}])
 
 (defn- transaction-details
   []
@@ -72,9 +74,11 @@
                                                    theme)))]
      [data-item
       {:title         (i18n/label :t/max-slippage)
-       :subtitle      max-slippage
+       :subtitle      (str max-slippage "%")
        :subtitle-icon :i/edit
-       :size          :small}]]))
+       :size          :small
+       :on-press      #(rf/dispatch [:show-bottom-sheet
+                                     {:content slippage-settings/view}])}]]))
 
 (defn- pay-token-input
   [{:keys [input-state on-max-press on-input-focus on-token-press on-approve-press input-focused?]}]

--- a/translations/en.json
+++ b/translations/en.json
@@ -2320,6 +2320,7 @@
   "slide-to-sign": "Slide to sign",
   "slide-to-swap": "Slide to swap",
   "slippage": "Slippage",
+  "slippage-cant-be-more-than-30": "Slippage can't be more than 30",
   "slippage-may-be-higher-than-necessary": "Slippage may be higher than necessary",
   "slippage-settings": "Slippage settings",
   "slippage-settings-description": "Your transaction will revert if the price changes more than the slippage percentage",

